ingestion/src/metadata/ingestion/source/dashboard/lightdash/metadata.py

```diff
@@ -0,0 +1,184 @@
+                        service_name=self.context.dashboard_service.fullyQualifiedName.__root__,
+                        chart_name=chart.name.__root__,
+                    )
+                    for chart in self.context.charts
+                ],
+                service=self.context.dashboard_service.fullyQualifiedName.__root__,
+            )
+            yield dashboard_request
+            self.register_record(dashboard_request=dashboard_request)
+        except Exception as exc:  # pylint: disable=broad-except
+            logger.debug(traceback.format_exc())
+            logger.warning(
+                f"Error creating dashboard [{dashboard_details.name}]: {exc}"
+            )
+
+    def yield_dashboard_chart(
+        self, dashboard_details: LightdashChart
+    ) -> Optional[Iterable[CreateChartRequest]]:
+        """Get chart method
+
+        Args:
+            dashboard_details:
+        Returns:
+            Iterable[CreateChartRequest]
+        """
+        charts = self.charts
+        for chart in charts:
+            try:
+                chart_url = (
+                    f"{clean_uri(self.service_connection.hostPort)}/question/{chart.uuid}-"
+                    f"{replace_special_with(raw=chart.name.lower(), replacement='-')}"
+                )
+                if filter_by_chart(self.source_config.chartFilterPattern, chart.name):
+                    self.status.filter(chart.name, "Chart Pattern not allowed")
+                    continue
+                yield CreateChartRequest(
+                    name=chart.uuid,
+                    displayName=chart.name,
+                    description=chart.description,
+                    sourceUrl=chart_url,
+                    service=self.context.dashboard_service.fullyQualifiedName.__root__,
+                )
+                self.status.scanned(chart.name)
+            except Exception as exc:  # pylint: disable=broad-except
+                logger.debug(traceback.format_exc())
+                logger.warning(f"Error creating chart [{chart}]: {exc}")
+
+    def yield_dashboard_lineage_details(
+        self,
+        dashboard_details: LightdashDashboard,
+        db_service_name: Optional[str],
+    ) -> Optional[Iterable[AddLineageRequest]]:
+        """Get lineage method
+
+        Args:
+            dashboard_details
+        """
+        chart_list, dashboard_name = (
+            dashboard_details.charts,
+            str(dashboard_details.uuid),
+        )
+        for chart in chart_list:
+            try:
+                yield from self._yield_lineage_from_query(
+                    chart_details=chart,
+                    db_service_name=db_service_name,
+                    dashboard_name=dashboard_name,
+                ) or []
+
+            except Exception as exc:  # pylint: disable=broad-except
+                logger.debug(traceback.format_exc())
+                logger.error(f"Error creating chart [{chart}]: {exc}")
```

Comment:
you can remove this for now and just leave the doc string. We can work on the implementation later.